table/src/main/java/com/bazaarvoice/emodb/table/db/astyanax/AstyanaxTableDAO.java

```diff
@@ -1394,9 +1469,76 @@ public void clearStashTokenRangeSnapshot(String stashId) {
         _stashTableDao.clearTokenRanges(stashId);
     }
 
+    @Timed (name = "bv.emodb.table.AstyanaxTableDAO.listMetadataChangedTables", absolute = true)
+    @Override
+    public Iterator<Map<String, Object>> listMetadataChangedTables(DateTime from, DateTime to, LimitCounter limit) {
+        checkArgument(limit.remaining() > 0, "Limit must be > 0");
+
+        final Iterator<Map<String, Object>> tableIter =
+                _backingStore.scan(_systemTableMetadataChanges, null, limit, ReadConsistency.STRONG);
```

Comment:
I don't recommend this approach.  With this approach over time the time to query grows with the amount of data in the metadata table, even if "from" and "to" is only a single day.  I'd instead recommend having logic which determines all of the days between from (inclusive) and to (exclusive), then querying each of those individually. I'd also put a hard limit on the maximum number of days they can query (30? 180? 365?) and throw an exception if the range between "from" and "to" is longer than that.

Reply:
My initial code had a monitor in place which deletes the old dated table records, so at any point the new system table would not have more than certain (~30) records and so the scan would have been OK, may be. But yeah, either way it's better to just get a record for each day separately as we cannot underestimate the delta size for each record.